### PR TITLE
refactored SecurityDesc

### PR DESCRIPTION
**Pull request by AndreasEhret**
- split RequestedPermissionLevel by AppletPermissionLevel and ApplicationPermissionLevel to better reflect JSR spec and entangle applet/html from application/jnlp logic
- improved documentation

### CODE REVIEW

**Comment by judovana** [comment]
looks good

File: core/src/main/java/net/adoptopenjdk/icedteaweb/jnlp/element/security/AppletPermissionLevel.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    /**
+     * Indicates that no applet permissions are specified.
+     * @deprecated
+     */
+    @Deprecated
+    // consider to handle the absence of an permissions param different
+    // as there is no such thing as NONE in the specs
+    NONE(null);
+
+    private String value;
```
Comment (hendrikebbers) [inline core/src/main/java/net/adoptopenjdk/icedteaweb/jnlp/element/security/AppletPermissionLevel.java]: final

File: core/src/main/java/net/adoptopenjdk/icedteaweb/jnlp/element/security/AppletPermissionLevel.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    AppletPermissionLevel(String value) {
```
Comment (hendrikebbers) [inline core/src/main/java/net/adoptopenjdk/icedteaweb/jnlp/element/security/AppletPermissionLevel.java]: final

File: core/src/main/java/net/adoptopenjdk/icedteaweb/jnlp/element/security/SecurityDesc.java [hunk 3]
```diff
@@ -299,21 +231,21 @@ public RequestedPermissionLevel fromHtmlString(final String htmlString) {
...
      * Create a security descriptor.
      *
      * @param file the JNLP file
-     * @param requestedPermissionLevel the permission level specified in the JNLP
+     * @param applicationPermissionLevel the permissions specified in the JNLP
      * @param type the type of security
      * @param downloadHost the download host (can always connect to)
      */
-    public SecurityDesc(JNLPFile file, RequestedPermissionLevel requestedPermissionLevel, Object type, URL downloadHost) {
+    public SecurityDesc(JNLPFile file, ApplicationPermissionLevel applicationPermissionLevel, Object type, URL downloadHost) {
```
Comment (hendrikebbers) [inline core/src/main/java/net/adoptopenjdk/icedteaweb/jnlp/element/security/SecurityDesc.java]: final

File: core/src/main/java/net/sourceforge/jnlp/JNLPFile.java [hunk 4]
```diff
@@ -1060,36 +1071,41 @@ public ManifestBoolean isTrustedLibrary() {
          * @return value of Permissions manifest attribute
          */
         public ManifestBoolean isSandboxForced() {
-            String s = getAttribute(PERMISSIONS);
-            if (s == null) {
+            String permissionLevel = getManifestPermissionsAttribute();
```
Comment (hendrikebbers) [inline core/src/main/java/net/sourceforge/jnlp/JNLPFile.java]: final

File: core/src/main/java/net/sourceforge/jnlp/JNLPFile.java [hunk 5]
```diff
@@ -1060,36 +1071,41 @@ public ManifestBoolean isTrustedLibrary() {
...
-
-
         }
         /**
          * http://docs.oracle.com/javase/7/docs/technotes/guides/jweb/manifest.html#permissions
          * @return plain string values of Permissions manifest attribute
          */
         public String permissionsToString() {
-            String s = getAttribute(PERMISSIONS);
+            String s = getManifestPermissionsAttribute();
```
Comment (hendrikebbers) [inline core/src/main/java/net/sourceforge/jnlp/JNLPFile.java]: final

File: core/src/main/java/net/adoptopenjdk/icedteaweb/jnlp/element/security/AppletPermissionLevel.java (hunk 1 above)
Comment (AndreasEhret) [reply core/src/main/java/net/adoptopenjdk/icedteaweb/jnlp/element/security/AppletPermissionLevel.java]: done

File: core/src/main/java/net/adoptopenjdk/icedteaweb/jnlp/element/security/AppletPermissionLevel.java (hunk 2 above)
Comment (AndreasEhret) [reply core/src/main/java/net/adoptopenjdk/icedteaweb/jnlp/element/security/AppletPermissionLevel.java]: done

File: core/src/main/java/net/adoptopenjdk/icedteaweb/jnlp/element/security/SecurityDesc.java (hunk 3 above)
Comment (AndreasEhret) [reply core/src/main/java/net/adoptopenjdk/icedteaweb/jnlp/element/security/SecurityDesc.java]: done

File: core/src/main/java/net/sourceforge/jnlp/JNLPFile.java (hunk 5 above)
Comment (AndreasEhret) [reply core/src/main/java/net/sourceforge/jnlp/JNLPFile.java]: done

File: core/src/main/java/net/sourceforge/jnlp/JNLPFile.java (hunk 4 above)
Comment (AndreasEhret) [reply core/src/main/java/net/sourceforge/jnlp/JNLPFile.java]: done

**Pull request merged**